webview-ui/src/ClusterProperties/AgentPoolDisplay.tsx

```diff
@@ -1,22 +1,54 @@
 import styles from "./ClusterProperties.module.css";
-import { AgentPoolProfileInfo } from "../../../src/webview-contract/webviewDefinitions/clusterProperties";
+import { AgentPoolProfileInfo, ClusterInfo } from "../../../src/webview-contract/webviewDefinitions/clusterProperties";
+import { VSCodeButton } from "@vscode/webview-ui-toolkit/react";
+import { EventDef, vscode } from "./state";
+import { EventHandlers } from "../utilities/state";
 
 export interface AgentPoolDisplayProps {
-    profileInfo: AgentPoolProfileInfo
+    clusterInfo: ClusterInfo;
+    profileInfo: AgentPoolProfileInfo;
+    eventHandlers: EventHandlers<EventDef>;
+    clusterOperationRequested: boolean;
 }
 
+// Abort on anything except: Canceled, Failed, Succeeded
+// Check about deleting
+type ProvisionState = "CapturingSecurityVHDSnapshot" | "Creating" | "Deleting" | "RefreshingServicePrincipalProfile"
+                    | "RotatingClusterCertificates" | "Scaling" | "Starting" | "Stopping" | "Succeeded" | "Updating" 
+                    | "Upgrading" | "UpgradingNodeImageVersion";
+
 export function AgentPoolDisplay(props: AgentPoolDisplayProps) {
+    const isProvisioningStateFromCluster = props.clusterInfo.provisioningState === props.profileInfo.provisioningState;
+    const isOperationInProgress = !["Canceled", "Failed", "Succeeded"].includes(props.profileInfo.provisioningState);
```

Comment:
If we replace the `ProvisionState` type with `terminalProvisioningStates`, we can change this to:
```typescript
const isOperationInProgress = !terminalProvisioningStates.includes(props.profileInfo.provisioningState);
```